src/api/controllers/IdentityController.js

```diff
@@ -284,6 +285,53 @@ class IdentityController extends PointSDKController {
 
         return this._response({eligibility, reason});
     }
+
+    async resolveDomain() {
+        const SUPPORTED_NAME_SERVICES = ['sns', 'ens'];
+        const SUPPORTED_TLD = ['.sol', '.eth'];
+
+        const {service, domain} = this.req.params;
+
+        if (!SUPPORTED_NAME_SERVICES.includes(service)) {
+            const status = 400;
+            const errorMsg = `Unsupported name service "${service}".`;
+            this.rep.status(status);
+            return this._status(status)._response({errorMsg});
+        }
+
+        const tld = SUPPORTED_TLD.find(tld => domain.endsWith(tld));
+        if (!tld) {
+            const status = 400;
+            const errorMsg = `Unsupported TLD in "${domain}".`;
+            this.rep.status(status);
+            return this._status(status)._response({errorMsg});
+        }
+
+        try {
+            let registry;
+            switch (service) {
```

Comment:
This `switch` is the kind of thing that we should probably avoid, having a thin abstraction over the providers. However, I'd like to make that change separately (I can create a subtask for that), in order to keep these changes as small and focused as possible.